lib/y2lan_restart_common.pm

```diff
@@ -0,0 +1,109 @@
+    select_console 'x11';
+    x11_start_program("xterm -geometry 155x50+5+5", target_match => 'xterm');
+    become_root;
+    # make sure that firewalld is stopped, or we have later pops for firewall activation warning
+    # or timeout for command 'ip a' later
+    if (((is_sle && sle_version_at_least('15')) or (is_leap && leap_version_at_least('15.0')))
+        and assert_script_run("systemctl show -p ActiveState firewalld.service | grep ActiveState=active"))
+    {
+        systemctl 'stop firewalld';
+        assert_script_run("systemctl show -p ActiveState firewalld.service | grep ActiveState=inactive");
+    }
+    # enable debug for detailed messages and easier detection of restart
+    assert_script_run 'sed -i \'s/DEBUG="no"/DEBUG="yes"/\' /etc/sysconfig/network/config';
+    type_string "journalctl -f|egrep -i --line-buffered 'shutting down|ifdown all' > journal.log &\n";
+    assert_script_run '> journal.log';    # clear journal.log
+}
+
+sub open_network_settings {
+    type_string "yast2 lan\n";
+    assert_screen 'yast2_lan', 100;       # yast2 lan overview tab
+}
+
+sub close_network_settings {
+    wait_still_screen;
+    send_key 'alt-o';
+    # new: warning pops up for firewall, alt-y for assign it to zone
+    assert_screen([qw(yast2-lan-restart_firewall_active_warning yast2_closed_xterm_visible yast2_lan_packages_need_to_be_installed)], 120);
+    if (match_has_tag 'yast2-lan-restart_firewall_active_warning') {
+        send_key 'alt-y';
+        wait_still_screen 1;
+        send_key 'alt-n';
+        wait_still_screen 1;
+        send_key 'alt-o';
+    }
+    elsif (match_has_tag 'yast2_lan_packages_need_to_be_installed') {
+        send_key 'alt-i';
+    }
+    assert_screen 'yast2_closed_xterm_visible', 120;    # ensure coming back to root console
+    type_string "\n\n";                                 # make space for better readability of the console
+}
+
+sub check_network_status {
+    my ($expected_status, $device) = @_;
+    $expected_status //= 'no_restart';
+    assert_script_run 'ip a';
+    if ("$device" eq 'bond') {
+        record_soft_failure 'bsc#992113';
+    }
+    else {
+        assert_script_run 'dig suse.com|grep \'status: NOERROR\'';    # test if conection and DNS is working
+    }
+    assert_script_run 'cat journal.log';                              # print journal.log
+    if ("$expected_status" eq 'restart') {
+        assert_script_run '[ -s journal.log ]';                       # journal.log size is greater than zero (network restarted)
+    }
+    elsif ((is_sle && !sle_version_at_least('15')) || (is_leap && !leap_version_at_least('15.0'))) {
+        assert_script_run '[ ! -s journal.log ]';
+    }
+    else {
+        # original: assert_script_run '[ ! -s journal.log ]' -> journal.log size is not greater than zero (network not restarted),
+        # but this is not working anymore because of bsc#1070578
+        record_soft_failure 'bsc#1070578' if script_run '[ ! -s journal.log ]';
```

Comment:
I see that bug is fixed now. Should we reopen the bug?

Reply:
for me that line doesn't make sense, basically if it is sle15 and not expecting restart it checks for an empty log for giving a soft failure (shouldn't have been checked for a greater than zero log when the error was present: [ -s journal.log ]. It looks like the "else" could be removed.

Reply:
So let's remove it, as bug is resolved and it doesn't look right.